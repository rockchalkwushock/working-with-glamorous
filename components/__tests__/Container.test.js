import { shallow, render, mount } from 'enzyme'

import { Container } from '../index'

describe('Container Component', () => {
  let ui
  test('should render to DOM', () => {
    ui = <Container />
    expect(ui.type.displayName).toEqual('StyledContainer')
    expect(shallow(ui)).toMatchSnapshotWithGlamor(`enzyme.shallow`)
    expect(mount(ui)).toMatchSnapshotWithGlamor(`enzyme.mount`)
    expect(render(ui)).toMatchSnapshotWithGlamor(`enzyme.render`)
  })

  test('should yield true when color prop is passed ', () => {
    ui = <Container parent />
    expect(ui.props.parent).toEqual(true)
    expect(shallow(ui)).toMatchSnapshotWithGlamor(`enzyme.shallow`)
    expect(mount(ui)).toMatchSnapshotWithGlamor(`enzyme.mount`)
    expect(render(ui)).toMatchSnapshotWithGlamor(`enzyme.render`)
  })
})
