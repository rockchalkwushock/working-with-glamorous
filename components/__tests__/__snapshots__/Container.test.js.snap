// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`enzyme.mount 1`] = `
.css-dye5wl,
[data-css-dye5wl] {
  -webkit-justify-content: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-box-direction: normal;
  -webkit-box-orient: vertical;
  -webkit-box-align: center;
  -webkit-align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  -ms-flex-align: center;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  margin: 0px;
  flex: 1 1 auto;
  justify-content: center;
  flex-direction: column;
  align-items: center;
}

<StyledContainer>
  <div
    className="css-dye5wl"
  />
</StyledContainer>
`;

exports[`enzyme.mount 2`] = `
.css-1chxodp,
[data-css-1chxodp] {
  -webkit-flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-box-direction: normal;
  -webkit-box-orient: vertical;
  -webkit-box-align: center;
  -ms-flex-align: center;
  -webkit-justify-content: center;
  -webkit-align-items: center;
  -ms-flex-direction: column;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  min-height: 100vh;
  max-width: 50em;
  margin: auto;
  background-color: #0266A2;
  justify-content: center;
  flex-direction: column;
  align-items: center;
}

<StyledContainer
  parent={true}
>
  <div
    className="css-1chxodp"
  />
</StyledContainer>
`;

exports[`enzyme.render 1`] = `
.css-dye5wl,
[data-css-dye5wl] {
  -webkit-justify-content: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-box-direction: normal;
  -webkit-box-orient: vertical;
  -webkit-box-align: center;
  -webkit-align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  -ms-flex-align: center;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  margin: 0px;
  flex: 1 1 auto;
  justify-content: center;
  flex-direction: column;
  align-items: center;
}

<div
  class="css-dye5wl"
/>
`;

exports[`enzyme.render 2`] = `
.css-1chxodp,
[data-css-1chxodp] {
  -webkit-flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-box-direction: normal;
  -webkit-box-orient: vertical;
  -webkit-box-align: center;
  -ms-flex-align: center;
  -webkit-justify-content: center;
  -webkit-align-items: center;
  -ms-flex-direction: column;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  min-height: 100vh;
  max-width: 50em;
  margin: auto;
  background-color: #0266A2;
  justify-content: center;
  flex-direction: column;
  align-items: center;
}

<div
  class="css-1chxodp"
/>
`;

exports[`enzyme.shallow 1`] = `
.css-dye5wl,
[data-css-dye5wl] {
  -webkit-justify-content: center;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-box-direction: normal;
  -webkit-box-orient: vertical;
  -webkit-box-align: center;
  -webkit-align-items: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  -ms-flex-align: center;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  margin: 0px;
  flex: 1 1 auto;
  justify-content: center;
  flex-direction: column;
  align-items: center;
}

<div
  className="css-dye5wl"
/>
`;

exports[`enzyme.shallow 2`] = `
.css-1chxodp,
[data-css-1chxodp] {
  -webkit-flex-direction: column;
  -webkit-box-pack: center;
  -ms-flex-pack: center;
  -webkit-box-direction: normal;
  -webkit-box-orient: vertical;
  -webkit-box-align: center;
  -ms-flex-align: center;
  -webkit-justify-content: center;
  -webkit-align-items: center;
  -ms-flex-direction: column;
  display: -webkit-box;
  display: -moz-box;
  display: -ms-flexbox;
  display: -webkit-flex;
  display: flex;
  min-height: 100vh;
  max-width: 50em;
  margin: auto;
  background-color: #0266A2;
  justify-content: center;
  flex-direction: column;
  align-items: center;
}

<div
  className="css-1chxodp"
/>
`;
